ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschematools.py

```diff
@@ -0,0 +1,145 @@
+            validation_mode = DEFAULT_VALIDATION_MODE
+
+        if not isinstance(validation_mode, str):
+            raise AnsibleActionFail("The argument 'validation_mode' must be a string")
+
+        if validation_mode not in VALID_VALIDATION_MODES:
+            raise AnsibleActionFail(f"Invalid value '{validation_mode}' for the argument 'validation_mode'. Must be one of {VALID_VALIDATION_MODES}")
+
+        self.validation_mode = validation_mode
+
+    def convert_data(self, data: dict) -> list[str]:
+        """
+        Convert data according to the schema (convert_types)
+        The data conversion is done in-place (updating the original "data" dict).
+
+        avd_schema.convert returns a generator, which we iterate through in handle_exceptions to perform the actual conversions.
+        """
+        if self.conversion_mode == "disabled":
+            return []
+
+        result_messages = []
+        exceptions = self.avdschema.convert(data)
+        if conversion_counter := self.handle_validation_exceptions(exceptions, self.conversion_mode):
+            result_messages.append(f"{conversion_counter} data conversions done to conform to schema.")
+
+        return result_messages
+
+    def validate_data(self, data: dict) -> list:
+        """
+        Validate data according to the schema
+
+        avd_schema.validate returns a generator, which we iterate through in handle_exceptions to perform the actual conversions.
+        """
+        if self.validation_mode == "disabled":
+            return []
+
+        result_messages = []
+        exceptions = self.avdschema.validate(data)
+        if validation_counter := self.handle_validation_exceptions(exceptions, self.validation_mode):
+            result_messages.append(f"{validation_counter} errors found during schema validation of input vars.")
+
+        return result_messages
+
+    def convert_and_validate_data(self, data: dict) -> dict:
+        """
+        Convert & Validate data according to the schema
+
+        Calls conversion and validation methods and gather resulting messages
+
+        Returns dict which can contain either or both of the following keys:
+        - failed: <bool>
+        - msg: <str with summarys messages from conversion and validation>
+
+        The return value should be updated on Ansible Action "result"
+        """
+        result = {}
+        result_messages = []
+
+        # Perform data conversions
+        result_messages.extend(self.convert_data(data))
+
+        # Perform validation
+        validation_messages = self.validate_data(data)
+        if validation_messages:
+            result_messages.extend(validation_messages)
+            if self.validation_mode == "error":
+                result["failed"] = True
+
+        if result_messages:
+            result["msg"] = " ".join(result_messages)
+
+        return result
+
+    def handle_validation_exceptions(self, exceptions: Generator, mode: str) -> int:
```

Comment:
```suggestion
    def handle_validation_exceptions(self, exceptions: Generator, mode: str) -> int:
        """
        Iterate through the Generator of exceptions.
        This method is actually where the content of the generator gets executed.
        
        It displays various messages depending on the `mode` parameter
        
        Returns:
        - counter: <int> the number of AristaAvdError in the exceptions Generator
        """
```

Reply:
Done in https://github.com/aristanetworks/ansible-avd/pull/2317/commits/dafa42ba59550da6873460b35d38219c1ed3ea07